streams/src/main/java/org/apache/kafka/streams/processor/internals/TransactionManager.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.errors.TimeoutException;
+import org.apache.kafka.common.errors.UnknownProducerIdException;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.errors.TaskMigratedException;
+import org.apache.kafka.streams.processor.TaskId;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.Future;
+
+public class TransactionManager {
+    private final Producer<byte[], byte[]> producer;
+    private final String applicationId;
+    private final TaskId taskId;
+    private final boolean eosAlphaEnabled;
+
+    // used when eosEnabled is true only
+    private boolean transactionInFlight = false;
+
+    public TransactionManager(final Producer<byte[], byte[]> producer) {
+        this(producer, null, null);
+    }
+
+    public TransactionManager(final Producer<byte[], byte[]> producer,
+                              final String applicationId,
+                              final TaskId taskId) {
+        if ((applicationId != null && taskId == null) ||
```

Comment:
The passed in task-id are used for two purposes: 1) define eosAlphaEnabled, 2) use in logging (btw there are still some place where we do not check taskId == null). And in https://github.com/apache/kafka/pull/8058 we no longer log the task-id in TaskMigrated since when it happens we would migrate all the tasks anyways. So we do not need it in the TaskMigrated exception.

So I'd suggest the following:

a) we keep a String for taskIds which would be a single task-id if the passed in `taskId != null` or default to "all owned active tasks" for logging purposes only.
b) we keep the eosEnabled checking the taskId as is.

And moving forward as we remove the alpha we would replace this parameter with a boolean only since it would always be `all owned active tasks`.